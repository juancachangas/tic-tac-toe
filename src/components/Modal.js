import React from 'react';


const Modal = ({ status, names }) => {
  let message = '';
  switch (status) {
    case 'winner-0': {
      message = `${names[0]} WINS`;
      break;
    }
    case 'winner-1': {
      message = `${names[1]} WINS`;
      break;
    }
    case 'draw': {
      message = 'Dang! It\'s a draw';
      break;
    }
    case 'wrong_move': {
      message = 'Sorry! This box was taken long ago';
      break;
    }
    default: {
      break;
    }
  }
  return (
    <article className={`Modal ${status}`}>
      <h1 className="big-banner">
        {message}
      </h1>
    </article>
  );
};

export default Modal;
