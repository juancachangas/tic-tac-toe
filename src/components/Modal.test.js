import React from 'react';
import { shallow } from 'enzyme';

import Modal from './Modal';

describe('Modal component is', () => {
  it('rendering default', () => {
    const wrapper = shallow(
      <Modal status="new" names={['juan', 'camilo']} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('rendering player 1 winner message', () => {
    const wrapper = shallow(
      <Modal status="winner-0" names={['juan', 'camilo']} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('rendering player 2 winner message', () => {
    const wrapper = shallow(
      <Modal status="winner-1" names={['juan', 'camilo']} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('rendering draw message', () => {
    const wrapper = shallow(
      <Modal status="draw" names={['juan', 'camilo']} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('rendering error message', () => {
    const wrapper = shallow(
      <Modal status="wrong_move" names={['juan', 'camilo']} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
