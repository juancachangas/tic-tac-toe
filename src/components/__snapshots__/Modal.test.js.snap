// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component is rendering default 1`] = `
<article
  className="Modal new"
>
  <h1
    className="big-banner"
  />
</article>
`;

exports[`Modal component is rendering draw message 1`] = `
<article
  className="Modal draw"
>
  <h1
    className="big-banner"
  >
    Dang! It's a draw
  </h1>
</article>
`;

exports[`Modal component is rendering error message 1`] = `
<article
  className="Modal wrong_move"
>
  <h1
    className="big-banner"
  >
    Sorry! This box was taken long ago
  </h1>
</article>
`;

exports[`Modal component is rendering player 1 winner message 1`] = `
<article
  className="Modal winner-0"
>
  <h1
    className="big-banner"
  >
    juan WINS
  </h1>
</article>
`;

exports[`Modal component is rendering player 2 winner message 1`] = `
<article
  className="Modal winner-1"
>
  <h1
    className="big-banner"
  >
    camilo WINS
  </h1>
</article>
`;
